### PR TITLE
refactoring: Cleanup StartRest()

### PR DESCRIPTION
**Pull request by DesWurstes**

### CODE REVIEW

**Comment by domob1812** [comment]
utACK 13938/commits/923b9f5172d5a8d2841fb52e187bd057570853ee

**Comment by MarcoFalke** [comment]
If my memory doesn't fail me, we did something similar only 2 weeks ago:  Return void instead of bool for functions that cannot fail #13774 

How many are there still left?

**Comment by MarcoFalke** [comment]
utACK 923b9f5172d5a8d2841fb52e187bd057570853ee

**Comment by DesWurstes** [comment]
Yes, I saw this one while reviewing that Pull Request. I don’t know if there are more of those.

**Comment by Empact** [comment]
utACK 923b9f5

File: src/init.cpp [hunk 1]
```diff
@@ -736,8 +736,8 @@ static bool AppInitServers()
     StartRPC();
     if (!StartHTTPRPC())
         return false;
-    if (gArgs.GetBoolArg("-rest", DEFAULT_REST_ENABLE) && !StartREST())
-        return false;
+    if (gArgs.GetBoolArg("-rest", DEFAULT_REST_ENABLE))
```
Comment (sipa) [inline src/init.cpp]: Nit: if you're modifying this code anyway, update it to follow the style guide and put braces around then clause which isn't on the same line.

**Review by sipa — COMMENTED** [review]
utACK 923b9f5172d5a8d2841fb52e187bd057570853ee

File: src/init.cpp (hunk 1 above)
Comment (DesWurstes) [reply src/init.cpp]: Done

**Comment by Empact** [comment]
re-utACK 2da54f5

**Comment by practicalswift** [comment]
utACK 2da54f5a66f91306c064d685bffcee2062b6ebd2

**Review by practicalswift — APPROVED** [review]

**Comment by MarcoFalke** [comment]
utACK 2da54f5

**Review by donaloconnor — APPROVED** [review]
utACK https://github.com/bitcoin/bitcoin/commit/2da54f5a66f91306c064d685bffcee2062b6ebd2

**Comment by fanquake** [comment]
utACK 2da54f5

**Comment by laanwj** [comment]
> If my memory doesn't fail me, we did something similar only 2 weeks ago: Return void instead of bool for functions that cannot fail #13774
> How many are there still left?

I think this is a good point, I'm also a bit divided on changes like this. It seems unnecessary.

A consistent convention for error handling makes sense and if a function **cannot** return an error that might mean that some potential errors go ignored, or do a lazy thing such as assert out.

All this needs to be re-added when proper error handling is introduced in the future.

**Comment by DesWurstes** [comment]
Maybe we should aim to replace exceptions with returning bool?

**Pull request merged**